### PR TITLE
Bugfix for fans not turning off when slider gets set 0

### PR DESCRIPTION
**Pull request by gabe565**
Just noticed that one of my last commits made it so that fans would not turn off when set to 0. This fixes that!

Also thanks for the shoutout!

### CODE REVIEW

**Comment by thomasloven** [comment]
Actually, I changed that because if the lowest speed of the fan is "off" (wemo, insteon, isy994, smartthings, tuya, zwave, zehnder, zigbee), then both the leftmost and the second leftmost option will be off.

This needs more work.

**Comment by gabe565** [comment]
Yeah that's a good point.

I had kind of assumed that fans behave similarly to lights and don't include an "off" option, but I am looking through all of the fan components right now and it looks like most do the opposite.

Your change is definitely the better choice. I am only using fan templates so I can pretty easily add an off option. This could be a new config item, don't know how often it would get used, though.

**Comment by thomasloven** [comment]
It seems the state of the off speed is always "off", so I guess we could check that, and use the appropriate method.

**Comment by gabe565** [comment]
What do you think about this change?

If the first entry is not `off`, it adds the off entry at slider level 0. Otherwise, it works exactly how it does now. Tested with both and they look to work either way for me.

Let me know if you see anything that could be changed/improved on.

File: slider-entity-row.js [hunk 1]
```diff
@@ -204,7 +204,10 @@ class SliderEntityRow extends Polymer.Element {
           return stateObj.attributes.speed;
         },
         min: (stateObj) => 0,
-        max: (stateObj) => stateObj.attributes.speed_list.length - 1,
+        max: (stateObj) => {
+          this.controller.valueDelta = stateObj.attributes.speed_list[0].toLowerCase() === 'off' ? 0 : 1;
```
Comment (thomasloven) [inline slider-entity-row.js]: This means `set` and `get` will fail if `max` hasn't been called first. While that's always true now, it's annoying to maintain...

Comment (gabe565) [reply slider-entity-row.js]: Hmm that's a good point. I could see that causing problems...     
I could set `valueDelta: 0` in the object there, but I would rather keep it similar to the other controller objects. I would prefer that to adding if statements all over the place though.

**Comment by thomasloven** [comment]
I'm starting to think perhaps it would be easier (or at least *better*) to fix the dyson, mqtt, template, wink and xiamo fans in Home Assistant to always report off as a speed...

**Comment by thomasloven** [comment]
Actually... with a quick look it seems that the fans that don't have "off" as a speed work that way by design (facepalm). As such, I think it's better to leave this as is, and let users who make mqtt or template fans manually add "off" as a speed if and when it makes sense to do so.

Thanks for your work anyway!

**Pull request closed without merging**